admin/settings-field-post-types-support.php

```diff
@@ -93,40 +93,64 @@ function ( $post_status_object ) {
 
 					$current = ( null === $post_expiration_info ) ? '' : $post_expiration_info->post_status;
 
-					$options = [
-						'' => '',
+					$groups = [
+						'public'     => [
+							'label'   => __( 'Public', 'pronamic-post-expiration' ),
+							'options' => [],
+						],
+						'non-public' => [
+							'label'   => __( 'Non Public', 'pronamic-post-expiration' ),
+							'options' => [],
+						],
 					];
 
 					foreach ( $post_status_objects as $key => $post_status_info ) {
-						$options[ $key ] = \sprintf(
-							/* translators: 1: Post status label, 2: Post status name, 3: Public yes/no. */
-							\__( '%1$s (%2$s) · Public: %3$s', 'pronamic-post-expiration' ),
+						$group = $post_status_info->public ? 'public' : 'non-public';
+
+						$groups[ $group ]['options'][ $key ] = \sprintf(
+							/* translators: 1: Post status label, 2: Post status name */
+							\__( '%1$s (%2$s)', 'pronamic-post-expiration' ),
 							\esc_html( $post_status_info->label ),
-							\esc_html( $post_status_info->name ),
-							\esc_html( $post_status_info->public ? \__( 'Yes', 'pronamic-post-expiration' ) : \__( 'No', 'pronamic-post-expiration' ) )
+							\esc_html( $post_status_info->name )
 						);
 					}
 
-					if ( ! \array_key_exists( $current, $options ) ) {
-						$options[ $current ] = $current;
-					}
-
 					$name = \sprintf(
 						'pronamic_post_expiration_config[post_types][%s][post_status]',
 						$post_type_info->name
 					);
 
 					?>
 					<select name="<?php echo \esc_attr( $name ); ?>" <?php \disabled( $disabled ); ?>>
+						<option value=""></option>
+
 						<?php
 
-						foreach ( $options as $value => $label ) {
-							printf(
-								'<option value="%s" %s>%s</option>',
-								\esc_attr( $value ),
-								\selected( $value, $current, false ),
-								\esc_html( $label )
-							);
+						foreach ( $groups as $group ) {
+							if ( 0 === count( $group['options'] ) ) {
+								continue;
+							}
```

Comment:
I prefer to use https://www.php.net/array_filter so that there is no such logic in the foreach.

Reply:
Changed in https://github.com/pronamic/pronamic-post-expiration/pull/3/commits/85d7052d0de94d9be7114aebc5e05ed5ada2e155.